### PR TITLE
Mango and misc

### PR DESCRIPTION
**Pull request by aenario**
This still needs more tests around edge cases, but opening a PR for @jinroh to review.

### CODE REVIEW

File: src/mango.js [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  return response.docs
+}
+
+// misc
+function capitalize (name) {
+  return name.charAt(0).toUpperCase() + name.slice(1)
+}
+
+function makeMapFunction (doctype, fields) {
+  if (Object.prototype.toString.call(fields) !== '[object Array]' ||
```
Comment (jinroh) [inline src/mango.js]: A reason not to use `Array.isArray` ?

Comment (aenario) [inline src/mango.js]: mostly FUD i guess :-)

Comment (aenario) [inline src/mango.js]: fixed in https://github.com/cozy/cozy-client-js/pull/9/commits/b4712a416c955b7c066da92d0626345e3c1ae1b1 which is part of #9
